lib/ex_unit/lib/ex_unit/capture_server.ex

```diff
@@ -59,14 +66,18 @@ defmodule ExUnit.CaptureServer do
     {:reply, :ok, release_device(ref, config)}
   end
 
-  def handle_call({:log_capture_on, pid}, _from, config) do
+  def handle_call({:log_capture_on, pid, string_io}, _from, config) do
     ref = Process.monitor(pid)
     refs = Map.put(config.log_captures, ref, true)
+    true = :ets.insert(@ets, {ref, string_io})
 
     if map_size(refs) == 1 do
+      handler_cfg = %{formatter: Logger.default_formatter()}
+
       status =
         with {:ok, config} <- :logger.get_handler_config(:default),
-             :ok <- :logger.remove_handler(:default) do
+             :ok <- :logger.remove_handler(:default),
+             :ok <- :logger.add_handler(@name, __MODULE__, handler_cfg) do
```

Comment:
We should attach this handler even if there is no default. So I recommend moving it out of the `with` and keeping these two code paths separated.

Reply:
We should also remember to remove this handler on maybe_add_console.

Reply:
maybe_add_console should probably be renamed to revert_handlers or something :D